07_errors/sirigithub/error.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import "fmt"
+
+type Error struct {
+	Message string
+	Code    int
+}
+
+const (
+	ErrInvalidValue   = 400
+	ErrIDNotFound     = 404
+	ErrMarshallData   = 400
+	ErrUrmarshallData = 400
+	ErrDatabaseConn   = 500
+	ErrDatabaseWrite  = 500
+	ErrDatabaseRead   = 500
+	ErrDatabseDelete  = 500
+)
+
+func (e *Error) Error() string {
+	return fmt.Sprintf("Error code %d: %s", e.Code, e.Message)
```

Comment:
with the `err` field, this would be `fmt.Sprintf("error code %d: %s: %v", e.Code, e.Message, e.err)`

Note that I've made the first letter lower case as per https://github.com/golang/go/wiki/CodeReviewComments#error-strings